src/ansi-c/c_typecheck_expr.cpp

```diff
@@ -2594,22 +2595,34 @@ exprt c_typecheck_baset::do_special_functions(
 
     // The value doesn't matter at all, we only care about the type.
     // Need to sync with typeclass.h.
-    const typet &type=follow(object.type());
-
-    unsigned type_number=
-      type.id()==ID_empty?0:
-      type.id()==ID_c_enum_tag?3:
-      (type.id()==ID_bool || type.id()==ID_c_bool)?4:
-      type.id()==ID_pointer?5:
-      type.id()==ID_floatbv?8:
-      (type.id()==ID_complex && type.subtype().id()==ID_floatbv)?9:
-      type.id()==ID_struct?12:
-      type.id()==ID_union?13:
-      type.id()==ID_array?14:
-      1; // int, short
-
-    // clang returns 15 for the three 'char' types,
-    // gcc treats these as 'int'
+    typet type=follow(object.type());
+
+    // use underlying type for bit fields
+    if(type.id()==ID_c_bit_field)
+      type=to_c_bit_field_type(type).subtype();
+
+    unsigned type_number;
+
+    if(type.id()==ID_bool || type.id()==ID_c_bool)
```

Comment:
Nit picking: clang-format style spacing around `==` and `=` (across this commit).

Reply:
done